src/test/kotlin/uk/gov/justice/digital/hmpps/courtregister/helper/FlywayRestoreExtension.kt

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.digital.hmpps.courtregister.helper
+
+import org.flywaydb.core.Flyway
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+class FlywayRestoreExtension : AfterAllCallback, BeforeEachCallback {
+  companion object {
+    @JvmField
+    var flyway: Flyway? = null
+  }
+
+  override fun beforeEach(context: ExtensionContext) {
+    val applicationContext = SpringExtension.getApplicationContext(context)
+    if (applicationContext.containsBean("flyway")) {
+      flyway = applicationContext.getBean(Flyway::class.java)
+    }
+  }
+
+  override fun afterAll(context: ExtensionContext) {
+    flyway?.clean()
+    flyway?.migrate()
```

Comment:
Not a great fan of tearing things down at the end of the test.  It means if someone runs a single test then they can't inspect the database at the end.  Could it be a `beforeAll` instead i.e. get the database into a known state before the test starts?

Reply:
Yes I agree.  I've basically used this as a sledgehammer solution so I didn't have to work out why tests are not tidying up after themselves.  I think I'll try and do it properly in the next PR?

Reply:
Whoops I misunderstood you there - I've actually gone for each test tidying up its own mess.  So it means you will have to stick a breakpoint on before the clean up and inspect the database at that point.  Not too onerous I don't think.

Reply:
yes, I more meant that if you move the `afterAll` to a `beforeAll` then you'll start each test in a known state and won't need any tidy up at all.  The database can then be inspected at the end if required without having to add breakpoints in etc.